File: picasso-commons/src/com/squareup/picasso/Picassos.java
package com.squareup.picasso;

import android.content.Context;
import android.support.annotation.NonNull;

/**
 * @author Hendra Anggrian (hendraanggrian@gmail.com)
 */
public final class Picassos {

    public static final String TAG = "Picassos";
    private static boolean DEBUG;

    private Picassos() {
    }

    @NonNull
    public static Cache getCache(@NonNull Context context) {
        return getCache(Picasso.with(context));
    }

    @NonNull
    public static Cache getCache(@NonNull Picasso picasso) {
        return picasso.cache;
    }

    public static void setDebug(boolean debug) {
        DEBUG = debug;
    }

    public static boolean isDebug() {
        return DEBUG;
    }
}

File: picasso-commons/src/com/squareup/picasso/Picassos.kt
package com.squareup.picasso

import android.content.Context

/**
 * @author Hendra Anggrian (hendraanggrian@gmail.com)
 */
object Picassos {
    internal val TAG = "Picassos"
    var isDebug: Boolean = false
}

fun Context.getPicasso(): Picasso = Picasso.with(this)
fun Picasso.getCache(): Cache = cache // memory cache, not storage